src/fsharp/absil/il.fs

```diff
@@ -2122,10 +2122,11 @@ type ILTypeDef(name: string, attributes: TypeAttributes, layout: ILTypeDefLayout
     member _.MethodImpls = methodImpls
     member _.Events = events
     member _.Properties = properties
+    member _.IsAttribute = isAttribute
```

Comment:
Storing `isAttribute: bool` might be a bit heavy weight for `ILTypeDef` as the question of `is the type an attribute` is answered before construction. Are we not able to determine this by looking at `extends`?

Reply:
Thinking about it, looking at `extends` won't give you all the information of the hierarchy. Seems the only reliable way of determining if it's an attribute is computing it before creating `ILTypeDef`. So, probably storing `isAttribute: bool` is fine.

Reply:
Just note that `ILTypeDef` tries to store data that is practically 1:1 with assembly metadata, which `isAttribute: bool` isn't a 1:1.

Reply:
> Thinking about it, looking at `extends` won't give you all the information of the hierarchy.

Yeah, that was a main reason of computing it for `ILTypeDef`.



> Because `IsAttribute` is only determined before writing an assembly, it does feel odd to have it on `ILTypeDef` as the type is for both reading and writing.
> 
> As a possible alternative, we could build some kind of storage in `IlxGen` that only hold the identities of types that are attributes. Then we can pass that storage to `ilwrite` in its `cenv`. That way those checks would look something like this: `if cenv.isAttrStorage.IsAttribute(td) ...`

Let me experiment with.